rust/lancedb/src/utils.rs

```diff
@@ -64,6 +82,21 @@ impl PatchReadParam for ReadParams {
     }
 }
 
+/// Validate table name.
+pub fn validate_table_name(name: &str) -> Result<()> {
+    if name.is_empty() {
+        return Err(Error::InvalidTableName {
+            name: name.to_string(),
+        });
+    }
+    if !TABLE_NAME_REGEX.is_match(name) {
+        return Err(Error::InvalidTableName {
+            name: name.to_string(),
+        });
```

Comment:
Maybe should give a reason here, so users know how to name their tables?
```suggestion
        return Err(Error::InvalidTableName {
            name: name.to_string(),
            reason: "Table names can only include letters, numbers, periods, and underscores.",
        });
```